internal/bminventory/inventory.go

```diff
@@ -258,6 +258,18 @@ type bareMetalInventory struct {
 	secretValidator validations.PullSecretValidator
 }
 
+func (b *bareMetalInventory) UpdateClusterInstallProgress(ctx context.Context, params installer.UpdateClusterInstallProgressParams) middleware.Responder {
+	var c common.Cluster
+
+	if err := b.db.First(&c, "id = ?", params.ClusterID).Error; err != nil {
```

Comment:
it can be 500 as well. anyway you can probably use a helper function https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L1766

Reply:
Done.